python/tvm/relay/transform/transform.py

```diff
@@ -261,32 +261,39 @@ def LazyGradientInit():
     return _ffi_api.LazyGradientInit()
 
 
-def FoldConstantExpr(expr, mod):
+def FoldConstantExpr(expr, mod, fold_qnn=False):
     """Fold the constant expressions in a Relay program.
     Parameters
     ----------
     expr: Expr
         The expression to fold
     mod: IRModule
         The module the expr lives in (for global calls)
+    fold_qnn: bool
+        Whether to fold constants for QNN operations.
 
     Returns
     -------
     new_expr: Expr
         The expr after Constant Folding
     """
-    return _ffi_api.FoldConstantExpr(expr, mod)
+    return _ffi_api.FoldConstantExpr(expr, mod, fold_qnn)
 
 
-def FoldConstant():
+def FoldConstant(fold_qnn=False):
     """Fold the constant expressions in a Relay program.
 
+    Parameters
+    ----------
+    fold_qnn: bool
```

Comment:
This should also be improved to state the reasoning you mentioned in the PR why all constant subgraphs of QNN is not folded by default.

Reply:
I guess the comment should be like next:
> Because of backward compatibility reason it skip qnn primitives from folding by default. There are some transformation passes like FakeQuantizationToInteger, which requires to keep qnn primitives for constant subgraphs. Uncontrolled constant folding of qnn primitives may break applicability of FakeQuantizationToInteger. We suggest to use FoldConstant pass with none default `fold_qnn` value only when all other qnn sensitive passes was already applied.

One more comment from my side. My initial thoughts was to use list of ignored operations instead of flag fold_qnn. Like next: `transform.FoldConstant(skip_ops=['qnn.*'])`

It may be more reusable approach if user will want to skip only particular primitive but not all qnn family. If you will find it helpful, we may change argument from flag to list.

Reply:
As for the description, I agree with content. Since you asked me :) : 
Maybe a general comment just highlight if a subsequent pass requires qnn dialect to be present even in constant sub-Expr, this flag should be set to false and cite the example ?

the second point, 

Im personally happy with a binary choice for now.

But I like the increased customizability of your suggestion (Should we stage it being skip_qnn_ops first and I guess highest scope would be all Legalizable ops). However, I dont immediately see a requirement for it. If we do that,  we'd need a bit more test coverage as well.

Maybe its better to keep it simple at this stage and consider a follow up ?

(It might better to get comments from the wider community as to the level of customization required. E.g. invidual qnn.ops , dialect of relay ops that get legalized (e.g. qnn) or individual op from all Legalizable ops)